lib/matplotlib/axes/_axes.py

```diff
@@ -3369,15 +3389,33 @@ def get_next_color():
                 texts.append(t)
 
             if autopct is not None:
-                xt = x + pctdistance * radius * math.cos(thetam)
-                yt = y + pctdistance * radius * math.sin(thetam)
+                if absolutefmt is not None:
+                    raise ValueError('Only one of autopct and absolutefmt may be used.')
                 if isinstance(autopct, str):
                     s = autopct % (100. * frac)
                 elif callable(autopct):
                     s = autopct(100. * frac)
                 else:
                     raise TypeError(
                         'autopct must be callable or a format string')
+                autolabels = True
+
+            elif absolutefmt is not None:
+                if isinstance(absolutefmt, str):
+                    s = absolutefmt % n
```

Comment:
It would be nice to support `absolutefmt.format(n)` here as well, and similarly for `autopct`  I vaguely remember seeing we had some logic _somewhere_ that could distinguish between old and new style format strings and apply them appropriately.  Did I dream that?  If not, could someone remind me where I saw it?

Reply:
I did not dream it
https://github.com/matplotlib/matplotlib/blob/183b04fb43f57ffe66da68c729a179e397cd35f6/lib/matplotlib/cbook.py#L2369-L2377